lib/ntbb-ladder.lib.php

```diff
@@ -224,7 +224,11 @@ function getTop() {
 			// 	$limit = 1000;
 			// }
 
-			$res = $ladderdb->query("SELECT * FROM `{$ladderdb->prefix}ladder` WHERE `formatid` = '{$this->formatid}' ORDER BY `elo` DESC LIMIT $limit");
+			if ($prefix) {
+				$res = $ladderdb->query("SELECT * FROM `{$ladderdb->prefix}ladder` WHERE `formatid` = '{$this->formatid}' AND `userid` LIKE '{$prefix}%' ORDER BY `elo` DESC LIMIT $limit");
```

Comment:
This is not an indexed query, and the ladder database is already at maximum stress despite zero unindexed queries, so I'm not particularly happy with this.

Ideally this would be done by manually flagging accounts as ladder-tour accounts, and then indexing by that flag.

Reply:
AIUI, there are 'multiple' ladder tours, even in OLT there is something like LT1...LT8 or whatever, and other metas etc can have their own tours/competitions (`bhlttar` is a current balanced hackmons ladder tour I think?). Is this something users would do at name registration? Is it something we automate based on a prefix? I don't know exactly how this scales.

I'm not against adding a tour bit to accounts - one use case for flagging accounts that came up recently is that **we could disable `/offertie` for ladder tours** in the same way we do on Smogtours/room tours. Similarly, being able to **flag an account as participating in a suspect test** would also make some Tier Leaders happy, given there was concern over how they would treat users who `/offertie`.

Reply:
Per discussion in chat I've switched to the overfetch + filter model. Does it need a `FORCE INDEX (formatid)` as well?

Reply:
In hindsight, it doesn't need a `FORCE INDEX`; it's only the replay database with broken indexes.